usr/lib/common/p11util.c

```diff
@@ -622,3 +623,33 @@ void p11_attribute_trim(CK_ATTRIBUTE *attr)
         }
     }
 }
+
+/* is_keytype_symmetric() - Check whether a key type corresponds to a
+ * symmetric key.
+ * @type key type to check
+ */
+CK_BBOOL is_keytype_symmetric(CK_KEY_TYPE type)
```

Comment:
You could also just check CK_OBJECT_CLASS for CKO_SECRET_KEY

Reply:
Well, if we have a check that only symmetric keys have CKO_SECRET_KEY I could.  I have not found that check.  But if I use a wrong key with AB wrap I get an error that I want to catch before.  Therefore I decided to use the actual key type instead of the object class.

Reply:
I don't see it anywhere explicitly stated in the PKCS#11 standard, but all places that I have seen where CKO_SECRET_KEY is used is dealing with symmetric keys. 
The PKCS#11 standard specifies for each key type _CKK_xxx the object class to use. E.g. for CKK_AES, it specifies that the class is CKO_SECRET_KEY .

> 2.8.2 AES secret key objects
> AES secret key objects (object class CKO_SECRET_KEY, key type CKK_AES) hold AES keys.

I guess if you lookup all the key types that you check here, you will see that all of them have  class CKO_SECRET_KEY . If that's the case, then a check for class = CKO_SECRET_KEY is sufficient in your code.

Reply:
Well, at the point of calling this function I only have a key type.  But since I have the template I can also extract the class.  At call side I basically need "Key is RSA or symmetric key".  So I based that purly on the key type.

Reply:
With these changes, the function becomes unused.  I will remove it from this commit.

Reply:
Done.